db/migrations/20170518101112_create_movie_locations.js

```diff
@@ -0,0 +1,13 @@
+'use strict';
+
+exports.up = function (Knex, Promise) {
+  return Knex.schema.createTable('locations_movies', (table) => {
+    table.integer('movie_id');
+    table.integer('location_id');
+    table.unique(['movie_id', 'location_id']);
```

Comment:
Instead of just making it `unique`, you should make it a [compound primary key](http://knexjs.org/#Schema-primary) (which is effectively a unique and a not null constraint).